src/Config.h

```diff
@@ -0,0 +1,1488 @@
+            _doc["hardware"]["sensors"]["scale"]["samples"] = SCALE_SAMPLES;
+            _doc["hardware"]["sensors"]["scale"]["type"] = 0;
+            _doc["hardware"]["sensors"]["scale"]["calibration"] = SCALE_CALIBRATION_FACTOR;
+            _doc["hardware"]["sensors"]["scale"]["calibration2"] = SCALE_CALIBRATION_FACTOR;
+            _doc["hardware"]["sensors"]["scale"]["known_weight"] = SCALE_KNOWN_WEIGHT;
+
+            // WiFi credentials flag
+            _doc["wifi"]["credentials_saved"] = false;
+        }
+
+        /**
+         * @brief Load configuration from file
+         *
+         * @return true if successful, false otherwise
+         */
+        bool load() {
+            if (!LittleFS.exists(CONFIG_FILE)) {
+                LOG(INFO, "Config file does not exist");
+
+                return false;
+            }
+
+            File file = LittleFS.open(CONFIG_FILE, "r");
+
+            if (!file) {
+                LOG(ERROR, "Failed to open config file for reading");
+
+                return false;
+            }
+
+            const DeserializationError error = deserializeJson(_doc, file);
+            file.close();
+
+            if (error) {
+                LOG(ERROR, "Failed to parse config file");
+                return false;
+            }
+
+            LOG(INFO, "Configuration loaded successfully");
+
+            return true;
+        }
+
+        /**
+         * @brief Save configuration to file
+         *
+         * @return true if successful, false otherwise
+         */
+        [[nodiscard]] bool save() const {
+            File file = LittleFS.open(CONFIG_FILE, "w");
+
+            if (!file) {
+                LOG(ERROR, "Failed to open config file for writing");
+                return false;
+            }
+
+            if (serializeJson(_doc, file) == 0) {
+                LOG(ERROR, "Failed to write config to file");
+                file.close();
+                return false;
+            }
+
+            file.close();
+            LOG(INFO, "Configuration saved successfully");
+
+            return true;
+        }
+
+        /**
+         * @brief Reset configuration to factory defaults
+         *
+         * @return true if successful, false otherwise
+         */
+        bool factoryReset() {
+            createDefaults();
+            return save();
+        }
+
+        bool validateAndApplyFromJson(const String& jsonString) {
+            DynamicJsonDocument doc(4096);
+            const DeserializationError error = deserializeJson(doc, jsonString);
+
+            if (error) {
+                LOGF(ERROR, "JSON parsing failed: %s", error.c_str());
+                return false;
+            }
+
+            if (!validateAndApplyConfig(doc)) {
+                return false;
+            }
+
+            return true;
+        }
+
+        // WiFi credentials flag
+        bool getWifiCredentialsSaved() {
+            return _doc["wifi"]["credentials_saved"] | false;
```

Comment:
I presume they are saved elsewhere (by the wifi manager?)?

Reply:
This is a bit of a relic. I kept it in place because some of the business logic in setup() was relying on it. Can probably be removed!